xaitk_saliency/impls/vis_sal_classifier/rise_scoring.py

```diff
@@ -0,0 +1,71 @@
+from typing import Dict, Any
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+from sklearn.preprocessing import maxabs_scale
+
+
+class RISEScoring (ImageClassifierSaliencyMapGenerator):
+    """
+    This saliency implementation transforms black-box
+    image classification scores into saliency heatmaps. This should
+    require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+
+    The perturbation masks used by the following implementation are
+    expected to be of type integer. Masks containing values of type
+    float are rounded to the nearest value and binarized
+    with value 1 replacing values greater than or equal to half of
+    the maximum value in mask after rounding while 0 replaces the rest.
+    """
+
+    def __init__(
+        self,
+        p1: float = 0.0,
+    ):
+        """
+        Generate RISE-based saliency maps with optional p1 de-biasing.
```

Comment:
We can also reference this de-biased RISE version:

```
Based on Hatakeyama et. al: https://openaccess.thecvf.com/content/ACCV2020/papers/Hatakeyama_Visualizing_Color-wise_Saliency_of_Black-Box_Image_Classification_Models_ACCV_2020_paper.pdf
```

Reply:
Added to the class-level doc-string.